hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -1418,7 +1418,13 @@ def group_by(group, agg_expr) -> DictExpression:
 
 @typecheck(expr=expr_any)
 def _prev_nonnull(expr) -> ArrayExpression:
-    return _agg_func('PrevNonnull', [expr], expr.dtype, [])
+    wrap = expr.dtype in {tint32, tint64, tfloat32, tfloat64, tbool, tcall}
```

Comment:
`is_primitive`?

Reply:
I tried using `is_primitive or == tcall` and that seemed worse.